libraries/incubator/ORM/Service/RepositoryFactory.php

```diff
@@ -0,0 +1,234 @@
+<?php
+/**
+ * Part of the Joomla Framework ORM Package
+ *
+ * @copyright  Copyright (C) 2015 - 2016 Open Source Matters, Inc. All rights reserved.
+ * @license    GNU General Public License version 2 or later; see LICENSE
+ */
+
+namespace Joomla\ORM\Service;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\DriverManager;
+use Joomla\ORM\Definition\Locator\Locator;
+use Joomla\ORM\Definition\Locator\Strategy\RecursiveDirectoryStrategy;
+use Joomla\ORM\Entity\EntityBuilder;
+use Joomla\ORM\Entity\EntityRegistry;
+use Joomla\ORM\Exception\OrmException;
+use Joomla\ORM\IdAccessorRegistry;
+use Joomla\ORM\Repository\Repository;
+use Joomla\ORM\Repository\RepositoryInterface;
+use Joomla\ORM\Storage\Csv\CsvDataGateway;
+use Joomla\ORM\Storage\Csv\CsvDataMapper;
+use Joomla\ORM\Storage\DataMapperInterface;
+use Joomla\ORM\Storage\Doctrine\DoctrineDataMapper;
+use Joomla\ORM\UnitOfWork\TransactionInterface;
+use Joomla\ORM\UnitOfWork\UnitOfWork;
+
+/**
+ * Repository Factory.
+ *
+ * @package Joomla/ORM
+ *
+ * @since   __DEPLOY_VERSION__
+ */
+class RepositoryFactory
+{
+	/** @var  array */
+	private $config;
+
+	/** @var  CsvDataGateway|Connection */
+	private $connection;
+
+	/** @var  array */
+	private $connections;
+
+	/** @var EntityBuilder The entity builder */
+	private $builder;
+
+	/** @var UnitOfWork The unit of work to use in the tests */
+	private $unitOfWork = null;
+
+	/** @var EntityRegistry The entity registry to use in tests */
+	private $entityRegistry = null;
+
+	/**
+	 * RepositoryFactory constructor.
+	 *
+	 * @api
+	 *
+	 * @param   array                      $config     The configuration
+	 * @param   CsvDataGateway|Connection  $connection The connection / gateway
+	 * @param   TransactionInterface       $transactor A Transactor
+	 */
+	public function __construct(array $config, $connection, $transactor)
```

Comment:
Why is the connection passed as parameter, it is built, when the entity is loaded, or?